### PR TITLE
[Feature] Remove tokens prop support

### PR DESCRIPTION
**Pull request by aappoalander**
## Description
This PR removes support for tokens prop for every component. In addition, it removes internal theme utils that handle merging custom and default tokens. This is done to have better performance and not recalculate styles on every render, but rather during build.

Some styles that depend on props were changed to use a class approach instead. Colors is now the only component that uses props for calculating styles and it is only used internally in styleguidist.

Some minor fixes for previously broken or missing styles and class names were also done.

## Motivation and Context
Merging custom and default tokens and calculating styles on every render caused poor performance on situations where there were a large number of components in one view.

## How Has This Been Tested?
Tested using Styleguidist, CRA TS and SSR test projects. In addition a performance test was done using a large number of components and a clean CRA TS setup. Tested using Mac OS and Win 10 with Chrome, Firefox, Safari and Edge + VoiceOver/NVDA.

There are only a couple of minor changes in the snapshots and none in the actual resulting styles.

## Release notes
- Breaking Change: Tokens prop is now removed from all components

### CODE REVIEW

File: src/core/Breadcrumb/Breadcrumb.baseStyles.ts [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
 import { nav, list, listItem, font } from '../theme/reset';
 
-export const baseStyles = withSuomifiTheme(
-  ({ theme }: TokensAndTheme) => css`
-    ${nav({ theme })}
-    ${font({ theme })('bodyTextSmall')}
+export const baseStyles = css`
+  ${nav(theme)}
+  ${font(theme)('bodyTextSmall')}
     background-color: ${theme.colors.whiteBase};
```
Comment (ketsappi) [inline src/core/Breadcrumb/Breadcrumb.baseStyles.ts:9]: Indentation doesn't match with others for the background-color.

Tried locally to change it to as others and ran linters and prettier; all seemed happy also then. 

_I remember we had similar things earlier also, but then there was something that linter/prettier did change it back to wrong indentation(?)_

File: src/core/Button/Button.tsx [hunk 2]
```diff
@@ -127,12 +123,8 @@ class BaseButton extends Component<ButtonProps> {
...
 }
 
-const StyledButton = styled(
-  ({ tokens, ...passProps }: ButtonProps & InternalTokensProp) => (
-    <BaseButton {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
+const StyledButton = styled((props: ButtonProps) => <BaseButton {...props} />)`
+  ${baseStyles}
```
Comment (ketsappi) [inline src/core/Button/Button.tsx:127]: In this and in similar places it seems to be possible to use simplified version as the props are not used like those were before.

Like this:
```
const StyledButton = styled(BaseButton)`
  ${baseStyles}
`;
```

File: src/core/Dropdown/Dropdown/Dropdown.test.tsx [hunk 3]
```diff
@@ -209,6 +208,5 @@ describe('Dropdown', () => {
 });
 
 test('CSS export', () => {
-  const css = cssFromBaseStyles(baseStyles);
-  expect(css).toEqual(expect.stringContaining('color'));
+  expect(baseStyles.join('')).toEqual(expect.stringContaining('color'));
 });
```
Comment (ketsappi) [inline src/core/Dropdown/Dropdown/Dropdown.test.tsx:212]: I vote to remove this whole test as it's value seems to be very low and vague.

File: src/core/Form/Textarea/Textarea.tsx [hunk 4]
```diff
@@ -22,15 +20,16 @@ const textareaClassNames = {
   fullWidth: `${baseClassName}--full-width`,
   textareaContainer: `${baseClassName}_textarea-element-container`,
   textarea: `${baseClassName}_textarea`,
+  resizeHorizontal: `${baseClassName}_textarea-resize--horizontal`,
+  resizeBoth: `${baseClassName}_textarea-resize--both`,
+  resizeNone: `${baseClassName}_textarea-resize--none`,
```
Comment (ketsappi) [inline src/core/Form/Textarea/Textarea.tsx:25]: This was good change (with all related changes) 👍

File: src/core/Form/Textarea/Textarea.baseStyles.tsx [hunk 5]
```diff
@@ -1,86 +1,94 @@
 import { css } from 'styled-components';
-import { TextareaProps } from './Textarea';
-import { withSuomifiTheme, TokensAndTheme } from '../../theme';
+import { defaultThemeTokens as theme } from '../../theme';
 import { disabledCursor } from '../../../components/utils/css';
 import { element, font } from '../../theme/reset';
 import { absoluteFocus } from '../../theme/utils/focus';
-import { Omit } from '../../../utils/typescript';
```
Comment (ketsappi) [inline src/core/Form/Textarea/Textarea.baseStyles.tsx:7]: This remove is good as TypeScript has the **Omit** nowadays. 

Maybe creating ticket to backlog to remove the use the custom Omit also in other places so we can get rid of it altogether.

File: src/core/Form/Toggle/ToggleButton/ToggeButton.baseStyles.tsx [hunk 6]
```diff
@@ -1,24 +1,25 @@
...
-        outline: 0;
-        & .fi-toggle_icon-container {
-          &:after {
-            ${absoluteFocus}
-            ${focusOverrides}
-          }
+export const baseStyles = css`
+  ${toggleBaseStyles}
+  &.fi-toggle--disabled > button {
+    ${disabledCursor}
```
Comment (ketsappi) [inline src/core/Form/Toggle/ToggleButton/ToggeButton.baseStyles.tsx:12]: Just remembered that in some other place we removed the use of this util and just told the style right away:

```css
cursor: not-allowed;
```

Maybe not the matter of this task, but creating ticket for backlog to get rid of this util?

File: src/core/StaticIcon/__snapshots__/StaticIcon.test.tsx.snap [hunk 7]
```diff
@@ -6,18 +6,31 @@ exports[`calling render with the same component on the same container does not r
   vertical-align: baseline;
 }
 
+<<<<<<< HEAD
```
Comment (ketsappi) [inline src/core/StaticIcon/__snapshots__/StaticIcon.test.tsx.snap:9]: This snapshot seem to have some Git stuff going 👀

File: src/core/StaticIcon/StaticIcon.tsx [hunk 8]
```diff
@@ -49,18 +57,18 @@ export class StaticIcon extends Component<StaticIconProps> {
...
     if (icon !== undefined) {
       return (
         <StyledSuomifiStaticIcon
           {...passProps}
-          className={classnames(baseClassName, className)}
           {...(!!highlightColor
             ? { highlightColor }
             : { highlightColor: tokens.colors.accentBase })}
           icon={icon}
+          className={getClassNames(this.props)}
```
Comment (ketsappi) [inline src/core/StaticIcon/StaticIcon.tsx:71]: Was there a certain benefit for having the classnames done in separate function?

File: src/core/LanguageMenu/LanguageMenu.test.tsx [hunk 9]
```diff
@@ -36,6 +35,5 @@ test(
 );
 
 test('CSS export', () => {
-  const css = cssFromBaseStyles(baseStyles);
-  expect(css).toEqual(expect.stringContaining('color'));
+  expect(baseStyles.join('')).toEqual(expect.stringContaining('color'));
```
Comment (ketsappi) [inline src/core/LanguageMenu/LanguageMenu.test.tsx:38]: Voting to get rid of this test also.

File: src/core/Breadcrumb/Breadcrumb.baseStyles.ts (hunk 1 above)
Comment (aappoalander) [reply src/core/Breadcrumb/Breadcrumb.baseStyles.ts:9]: Will fix, this are left untouched by linters/prettier.

File: src/core/Button/Button.tsx (hunk 2 above)
Comment (aappoalander) [reply src/core/Button/Button.tsx:127]: Good point here as well. Not sure if there are any perf implications, but it's better to have it cleaner.

File: src/core/Form/Textarea/Textarea.tsx (hunk 4 above)
Comment (aappoalander) [reply src/core/Form/Textarea/Textarea.tsx:25]: Might be needed elsewhere as well, but gets complicated quickly when there are a lot of options.

File: src/core/StaticIcon/__snapshots__/StaticIcon.test.tsx.snap (hunk 7 above)
Comment (aappoalander) [reply src/core/StaticIcon/__snapshots__/StaticIcon.test.tsx.snap]: Fixed already. Bad rebase.

File: src/core/Dropdown/Dropdown/Dropdown.test.tsx (hunk 3 above)
Comment (aappoalander) [reply src/core/Dropdown/Dropdown/Dropdown.test.tsx:212]: Removed from both Dropdown and Language menu.

File: src/core/Form/Textarea/Textarea.baseStyles.tsx (hunk 5 above)
Comment (aappoalander) [reply src/core/Form/Textarea/Textarea.baseStyles.tsx:7]: There were only three cases left. Those are now removed with the definition.

File: src/core/Form/Toggle/ToggleButton/ToggeButton.baseStyles.tsx (hunk 6 above)
Comment (aappoalander) [reply src/core/Form/Toggle/ToggleButton/ToggeButton.baseStyles.tsx:12]: Was such a small effort that replaced all with `cursor: not-allowed;` and removed the whole tooling. These kind of tools with only one function and no updates in the foreseeable future should be removed. Unnecessary abstractions.

File: src/core/StaticIcon/StaticIcon.tsx (hunk 8 above)
Comment (aappoalander) [reply src/core/StaticIcon/StaticIcon.tsx:66]: Removed and logic now in JSX.

File: src/core/LanguageMenu/LanguageMenu.test.tsx (hunk 9 above)
Comment (aappoalander) [reply src/core/LanguageMenu/LanguageMenu.test.tsx:38]: Removed.

File: src/core/Button/Button.tsx (hunk 2 above)
Comment (aappoalander) [reply src/core/Button/Button.tsx:127]: These should be now better. Did some refactoring in few places and moved AutoId to main level. The ones with className logic in styled were left untouched.

File: src/core/Breadcrumb/Breadcrumb.baseStyles.ts (hunk 1 above)
Comment (aappoalander) [reply src/core/Breadcrumb/Breadcrumb.baseStyles.ts]: Fixed all I could find.

**Review by LJKaski — APPROVED** [review]
Tested in Chrome, Edge and FF with styleguidist, DS-site and and CRA project + using NVDA just in case. Styles seem to work as they should.

**Pull request merged**